### PR TITLE
Add order attribution GA4 UTM parameters

### PR DESCRIPTION
**Pull request by layoutd**
### Changes proposed in this Pull Request:

This PR adds the 3 new UTM parameters captured by default in Sourcebuster, and stores with with the order data. They're also presented in the dropdown on the order edit for when present.

The parameters are:
- `utm_source_platform`
- `utm_creative_format`
- `utm_marketing_tactic`

See the Sourcebuster PR for more information: https://github.com/woocommerce/sourcebuster-js/pull/7

### How to test the changes in this Pull Request:

Summary: visit the shop with the new UTM parameters, complete a checkout. Confirm no errors, data is available in JS console, data is visible in Orders edit page. Confirm fields don't appear for purchases without those parameters. Test with both checkout types.
1. Visit the shop with a URL with the new parameters, like `site.com/shop/?utm_source=github_pr_test&utm_medium=cpc&utm_campaign=wc-testing&utm_term=SpécialChärs🐶&utm_content=ad1&utm_id=20110927&utm_source_platform=gads&utm_creative_format=img&utm_marketing_tactic=rkt`
2. Check that all the params are captured:
    In the dev console, execute `wc_order_attribution.getAttributionData()` and confirm that all the UTM parameters are present and have the correct value in the result (especially these 3 new params):
    image
4. Complete checkout, then view the order in the admin panel.
5. Confirm the new UTM parameter values are visible in the expanded **Details** values:
    image
6. In a new browser session, visit with a subset of UTM params and make a purchase (`site.com/shop/?utm_source=github_pr_test2`)
7. Confirm that the new parameters are no longer present in the dropdown
6. Change the checkout type (Classic↔Block) and repeat all steps (since attribution data is sent differently depending on checkout type).
    image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/templates/order/attribution-details.php [hunk 1]
```diff
@@ -82,6 +82,33 @@
...
 		<?php endif; ?>
 
+		<?php if ( array_key_exists( 'utm_source_platform', $meta ) ) : ?>
+		<h4>
+			<?php esc_html_e( 'Source platform', 'woocommerce' ); ?>
+		</h4>
+		<span class="order-attribution-utm-source-platform">
+				<?php echo esc_html( $meta['utm_source_platform'] ); ?>
+			</span>
+		<?php endif; ?>
```
Comment (ecgan) [inline plugins/woocommerce/templates/order/attribution-details.php]: The formatting isn't quite right here?

```suggestion
		<?php if ( array_key_exists( 'utm_source_platform', $meta ) ) : ?>
			<h4>
				<?php esc_html_e( 'Source platform', 'woocommerce' ); ?>
			</h4>
			<span class="order-attribution-utm-source-platform">
				<?php echo esc_html( $meta['utm_source_platform'] ); ?>
			</span>
		<?php endif; ?>
```

**Review by ecgan — APPROVED** [review]
Tested, works as expected, LGTM 🙂 👍

File: plugins/woocommerce/templates/order/attribution-details.php (hunk 1 above)
Comment (ecgan) [reply plugins/woocommerce/templates/order/attribution-details.php]: Note that this has been fixed in [`0f489ff` (#47639)](https://github.com/woocommerce/woocommerce/pull/47639/commits/0f489ffba0a5150952c53cf45306e678c9a5449e).

**Comment by layoutd** [comment]
Thanks @ecgan!

**Pull request merged**